### PR TITLE
added encryption for s3

### PR DESCRIPTION
**Pull request by bharat199708**
checks that s3 buckets are encrypted or not and list and list out all the unencrypted buckets

### CODE REVIEW

File: audit_config.yaml [hunk 1]
```diff
@@ -16,7 +16,14 @@ S3_PUBLIC_POLICY_GETOBJECT_ONLY:
   severity: Info
   is_global: True
   group: S3
-
+  
+S3_ENCRYPTION:
+  title: s3 encryption is not enabled
```
Comment (0xdabbad00) [inline audit_config.yaml:21]: Need to capitalize S3

File: audit_config.yaml [hunk 2]
```diff
@@ -16,7 +16,14 @@ S3_PUBLIC_POLICY_GETOBJECT_ONLY:
...
+  description: encryption is not done on these buckets
```
Comment (0xdabbad00) [inline audit_config.yaml:22]: I would reword to "S3 bucket encryption is not enabled"

File: shared/audit.py [hunk 3]
```diff
@@ -165,7 +165,16 @@ def audit_s3_buckets(findings, region):
                     },
                 )
             )
-
+            
+ def audit_encrypt_s3(findings, region):
```
Comment (0xdabbad00) [inline shared/audit.py:169]: This would be better added as part of the `audit_s3` function which checks the policy and ACL already.

**Review by 0xdabbad00 — CHANGES_REQUESTED** [review]
Thank you for the PR @bharat199708! I had a couple of minor changes. I can merge and edit these in if you like, or it'd be great if you could make these changes directly.

File: audit_config.yaml [hunk 4]
```diff
@@ -16,7 +16,14 @@ S3_PUBLIC_POLICY_GETOBJECT_ONLY:
...
+  severity: Medium
```
Comment (0xdabbad00) [inline audit_config.yaml:23]: I would set this as Low severity, as the threat is generally just if someone breaks into an AWS facility and steals the hard-drive. There are additional security benefits if KMS encryption is used with a custom key, but because this only checks if the bucket is encrypted generally, I wouldn't set the severity as high.

**Comment by bharat199708** [comment]
yes did those changes .i think we should also add encryption of EBS and RDS snapshots also ,
we should make it SOC 2compliance

File: audit_config.yaml (hunk 2 above)
Comment (bharat199708) [reply audit_config.yaml]: done

File: shared/audit.py (hunk 3 above)
Comment (bharat199708) [reply shared/audit.py:169]: I think encryption is totally different part , so we should add it as different function and other information related to encryption should be added there

Comment (KevinHock) [reply shared/audit.py:169]: I think we should add this to `audit_s3_buckets` as well. 

IMO it would kind of be nice to break out the body of `audit_s3_buckets` to call something like `check_s3__bucket_acl`/ `check_s3__bucket_policy`/` `check_s3__bucket_encryption`, instead of the e.g. `Check ACL`  comments.

File: audit_config.yaml (hunk 2 above)
Comment (KevinHock) [reply audit_config.yaml]: `description` isn't as explanatory as the other S3 checks, Scott probably has a better idea of what the description should be than I do 🤔

**Comment by 0xdabbad00** [comment]
Closing as stale.
